app/tests/api/logging/test_audit.py

```diff
@@ -2,63 +2,167 @@
 # Tests for api.logging.audit.
 #
 
+import io
 import logging
+import os
+import pathlib
+import socket
+import sys
+import urllib.request
+from typing import Any, Callable
+
+import pytest
 
 import api.logging.audit as audit
 
 
-def test_audit_hook(caplog):
-    caplog.set_level(logging.INFO)
+@pytest.fixture(scope="session")
+def init_audit_hook():
+    audit.init()
 
-    # Should appear in audit log.
-    audit.audit_hook("open", ("/dev/null", None, 123000))
 
-    # Various common cases that should not appear in audit log (normal behaviour & too noisy).
-    audit.audit_hook("compile", (b"def _(): pass", "<unknown>"))
-    audit.audit_hook("open", ("/srv/api/__pycache__/status.cpython-310.pyc", "r", 500010))
-    audit.audit_hook("os.chmod", (7, 1, -1))
-    audit.audit_hook(
-        "open", ("/app/.venv/lib/python3.10/site-packages/pytz/zoneinfo/US/Eastern", "r", 524288)
-    )
+test_audit_hook_data = [
+    pytest.param(eval, ("1+1", None, None), [{"msg": "exec"}], id="eval"),
+    pytest.param(exec, ("1+1", None, None), [{"msg": "exec"}], id="exec"),
+    pytest.param(
+        open,
+        ("/dev/null", "w"),
+        [
+            {
+                "msg": "open",
+                "audit.args.path": "/dev/null",
+                "audit.args.mode": "w",
+                "audit.args.flags": 524865,
+            }
+        ],
+        id="open",
+    ),
+    pytest.param(
+        io.open,
+        ("/dev/null", "w"),
+        [
+            {
+                "msg": "open",
+                "audit.args.path": "/dev/null",
+                "audit.args.mode": "w",
+                "audit.args.flags": 524865,
+            }
+        ],
+        id="io.open",
+    ),
```

Comment:
`io.open` is an alias for `open` so maybe we can remove this case. https://docs.python.org/3/library/io.html#io.open